### PR TITLE
feat: pay 0-amount invoices

### PR DESCRIPTION
**Pull request by rolznz**
Fixes https://github.com/getAlby/hub/issues/75

Currently only implemented in LDK. Includes a new screen for entering the amount (the same as the LNURL one, just without the ability to set a comment)

- [x] Update to use https://github.com/getAlby/js-lightning-tools/pull/181 (currently I have this branch linked locally)
- [x] Add support in LND LNClient - or could move to a separate issue

Tested by running 2 nodes locally and generating a 0-amount invoice from one and paying with the other. NOTE: there is no plan to add support for creating 0-amount invoices.

### CODE REVIEW

**Comment by bumi** [comment]
you know how much I like zero-amount invoices :D 

will this work with the supported backends or will this need more `if` ?

File: http/http_service.go [hunk 1]
```diff
@@ -438,7 +438,14 @@ func (httpSvc *HttpService) balancesHandler(c echo.Context) error {
 func (httpSvc *HttpService) sendPaymentHandler(c echo.Context) error {
 	ctx := c.Request().Context()
 
-	paymentResponse, err := httpSvc.api.SendPayment(ctx, c.Param("invoice"))
+	var payInvoiceRequest api.PayInvoiceRequest
```
Comment (im-adithya) [inline http/http_service.go:441]: Why not push the invoice to the `PayInvoiceRequest` struct?

**Review by im-adithya — APPROVED** [review]
Small nit regarding payload struct, otherwise tACK 👌

File: http/http_service.go (hunk 1 above)
Comment (im-adithya) [reply http/http_service.go:441]: Hmmm, true should be good 👍

File: frontend/package.json [hunk 2]
```diff
@@ -20,8 +20,8 @@
     "prepare": "cd .. && husky frontend/.husky"
   },
   "dependencies": {
-    "@getalby/lightning-tools": "^5.1.1",
     "@getalby/bitcoin-connect-react": "^3.6.2",
+    "@getalby/lightning-tools": "5.1.2",
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply frontend/package.json:24]: Ah true, sorry for this!

**Comment by SoulReaver-SS** [comment]
"NOTE: there is no plan to add support for creating 0-amount invoices." why?
That's where the meat is w/ 0 amount invoices, ability to generate them to receive flexible amounts in payment.

**Comment by rolznz** [comment]
@SoulReaver-SS we believe lightning addresses are a better solution than 0-amount invoices

**Pull request merged**